src/frontend/e2e/pages/filter.ts

```diff
@@ -1,103 +1,101 @@
-      }
-      if (filterData.collectionDateTo !== undefined) {
-        await page
-          .locator("input[name='collectionDateEnd']")
-          .fill(filterData.collectionDateTo);
-      }
+    await page
+      .locator(`text=${applyUploadDateSelector} >> [data-testid="button"]`)
+      .click();
+  }
+  // select upload date period
+  if (filterData.uploadDatePeriod !== undefined) {
+    await page.locator("button[label='Upload Date']").click();
+    await page.locator(`text=${filterData.uploadDatePeriod}`).nth(0).click();
+  }
+  // fill in collection date(s)
+  if (
+    filterData.collectionDateFrom !== undefined ||
+    filterData.collectionDateTo !== undefined
+  ) {
+    await page.locator("button[label='Collection Date']").click();
+    if (filterData.collectionDateFrom !== undefined) {
       await page
-        .locator(
-          "//div[not(contains(@style,'visibility: hidden')) and contains(@class,'MuiPaper-root')]/descendant::button[text()='Apply']"
-        )
-        .click();
+        .locator("input[name='collectionDateStart']")
+        .fill(filterData.collectionDateFrom);
     }
-    // select collection date period
-    if (filterData.collectionDatePeriod !== undefined) {
-      await page.locator("button[label='Collection Date']").click();
+    if (filterData.collectionDateTo !== undefined) {
       await page
-        .locator("div:not([style*='hidden'])[class*='MuiPaper-elevation'] li")
-        .filter({ hasText: filterData.collectionDatePeriod })
-        .click();
+        .locator("input[name='collectionDateEnd']")
+        .fill(filterData.collectionDateTo);
     }
-    // select lineage
-    if (filterData.lineage !== undefined) {
-      await page.locator("button[label='Lineage']").click();
-      for (const singleLineage of filterData.lineage) {
-        await page.locator("div[role='tooltip'] input").fill(singleLineage);
-        await page
-          .locator("ul[role='listbox']  .primary-text > div", {
-            hasText: singleLineage,
-          })
-          .first()
-          .click();
-      }
-      await page.keyboard.press("Escape"); //dismiss form
+    await page
+      .locator(`text=${applyCollectionDateSelector} >> [data-testid="button"]`)
+      .click();
+
+    //dismiss form
+    await page.keyboard.press("Escape");
+  }
+  // select collection date period
+  if (filterData.collectionDatePeriod !== undefined) {
+    await page.locator("button[label='Collection Date']").click();
+    await page
+      .locator(`text=${filterData.collectionDatePeriod}`)
+      .nth(1)
+      .click();
+  }
+  // select lineage(s)
+  if (filterData.lineage !== undefined) {
+    await page.locator('button:has-text("Lineage")').click();
+    await page
+      .locator('text=Search for a location​ >> [placeholder="Search"]')
+      .click();
+    for (const singleLineage of filterData.lineage) {
+      await page
+        .locator('text=Search for a location​ >> [placeholder="Search"]')
```

Comment:
I don't understand this selector. What is the double angle bracket representing here?

Reply:
Also, couldn't this be a `fillByPlaceholder`?